eureka-protempa-etl/src/main/java/edu/emory/cci/aiw/cvrg/eureka/etl/dao/JpaJobDao.java

```diff
@@ -152,6 +164,15 @@ public JpaJobDao(final Provider<EntityManager> inEMProvider) {
 						root.join(JobEntity_.jobEvents).get(JobEvent_.status),
 						jobFilter.getState()));
 			}
+			LOGGER.debug("Checking for recent.");
+			if (jobFilter.getRecent()) {
+				LOGGER.debug("Get recent job");
+				CriteriaQuery<Date> maxQuery = builder.createQuery(Date.class);
+				Root<JobEntity> rootForMax = maxQuery.from(JobEntity.class);
+				maxQuery.select(builder.greatest(rootForMax.get(JobEntity_.created)));
+				Date maxCreatedDate = entityManager.createQuery(maxQuery).getSingleResult();
```

Comment:
Try a nested query.

> On Jun 6, 2016, at 11:11 AM, Akshatha Pai <notifications@github.com> wrote:
> 
> In eureka-protempa-etl/src/main/java/edu/emory/cci/aiw/cvrg/eureka/etl/dao/JpaJobDao.java:
> 
> > @@ -152,6 +164,15 @@ public JpaJobDao(final Provider<EntityManager> inEMProvider) {
> >  						root.join(JobEntity_.jobEvents).get(JobEvent_.status),
> >  						jobFilter.getState()));
> >  			}
> > +			LOGGER.debug("Checking for recent.");
> > +			if (jobFilter.getRecent()) {
> > +				LOGGER.debug("Get recent job");
> > +				CriteriaQuery<Date> maxQuery = builder.createQuery(Date.class);
> > +				Root<JobEntity> rootForMax = maxQuery.from(JobEntity.class);
> > +				maxQuery.select(builder.greatest(rootForMax.get(JobEntity_.created)));
> > +				Date maxCreatedDate = entityManager.createQuery(maxQuery).getSingleResult();
> yes this is executing a query just to get the max date which is then used in the where clause to get the recent job entity. all examples of this online did something like this. i tried other ways of doing this as well but only this way seemed to work. Do you have any suggestions?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>